Reorganize getting-started + added a new tab "Core Concepts" (#24)
php-etl/documentation: content/core-concept/satellite/_index.en.md

version: '0.3'
satellites:
   - filesystem:
#       path: <path/to/folder>
        path: build
#...
{{< /tab >}}

{{< /tabs >}}

### Configure composer 

#### Dependencies

It's possible to declare the composer dependencies that our microservice needs with the `composer` key.

> Tip : This part is not mandatory. If you do not configure it, these packages (`php-etl/pipeline-contracts`,
`php-etl/pipeline`, `php-etl/pipeline-console-runtime`, `php-etl/workflow-console-runtime`, 
`psr/log`, `monolog/monolog`, `symfony/console`, `symfony/dependency-injection`) will be installed automatically.

> The `require` parameter allows to add all the packages, write like `package_name:version`, that we need for your microservice.

Reviewer: ```suggestion
The `require` parameter allows to add all the packages, written as `package_name:version`, that we need for your microservice.
```

et un coup on parle de "we", puis de "you" ?? qui écrit la config et qui a le microservice ?

Author: Il faudrait utiliser le "vous" plutôt que le "nous" 

J'ai changé la phrase pour avoir :

"The `require` parameter allows to add all the packages, written as `package_name:version`, that your microservice needs."

{{< /tabs >}}

#### Autoload

The `autoload` option is optional and allows you to configure your autoloader by specifying one or more namespaces and 
directories paths as if you were directly in the composer.json.

> Chaque autoload doit être constitué des clés suivantes :

Reviewer 2: ```suggestion
Every autoloading configuration shall be in the following format:
```

            paths: [""]
``` 

#### From local

> The `from_local` parameter is optional and copies local `composer.json`, `composer.lock` and `vendor` files in your 

Reviewer: ```suggestion
The `from_local` parameter is optional and copies local `composer.json`, `composer.lock` and `vendor` files into your 
```
pour faire comprendre qu'on le copie du local vers le microservice

Author: C'est corrigé

      from_local: true
``` 

#### Repositories

> The `repositories` option is optional and allow you to use directories that are hosted on other repositories than Packagist.

Reviewer 2: ```suggestion
The `repositories` parameter is optional. It allows you to use repositories that are not hosted on packagist.org.
```

Each repository should have the following configuration fields:

- `name`: the name of your repository
- `url`: the url of your repository

> - `type`: it allows to determine which authentication type is used to access your repo, see [Authentication principles](https://getcomposer.org/doc/articles/authentication-for-private-packages.md#authentication-principles).

Reviewer 2: **❔ question:** Tu es sûr que la phrase est au bon endroit ? Pour moi le type sert à indiquer si c'est un dossier, un repo git ou un repos composer. L'authentification est au chapitre suivant.‏

Author: Effectivement, c'est une erreur de ma part, je l'ai modifié

```yaml
version: '0.3'
satellites:
  - # ...
    composer:
      repositories:
        - { name: 'private-packagist', type: 'composer', url: 'https://repo.packagist.com/package/' }
``` 

#### Auth

> The `auth` option is optional and allows you to use packages that are protected by your server.

Reviewer 2: ```suggestion
The `auth` parameter is optional and allows you to use registries that are not public and accessed through an authentication.
```

> You will need to tell Composer how to authenticate to the server that hosts them.

Reviewer 2: ```suggestion
The option is the way for you to tell composer how to authenticate to the registry server.
```

Each auth can have the following configuration fields:
- `url`: the url of your repository
- `token`: when you use a connection via token, you must use this field

> > Notice : Actually, the only way to identify to a repository is to use tokens.

Reviewer 2: ```suggestion
> Notice : Currently, the only way to identify to a repository is to use tokens. Support for other authentication methods is in our backlog.
```

      from_local: true
``` 

#### Repositories

The `repositories` parameter is optional. It allows you to use repositories that are not hosted on [packagist.org](https://packagist.org/).

Each repository should have the following configuration fields:

- `name`: the name of your repository
- `url`: the url of your repository
- `type`: the type of your repository (vcs, composer, package, etc...)

```yaml
version: '0.3'
satellites:
  - # ...
    composer:
      repositories:

>         - { name: 'private-packagist', type: 'composer', url: 'https://repo.packagist.com/package/' }

Reviewer: on peut les mettre dans l'ordre enoncé juste au dessus : name, url, type

Author: J'ai modifié l'ordre

``` 

#### Auth

> The `auth` parameter is optional and allows you to use registries that are not public and accessed through an authentication.

Reviewer: ```suggestion
The `auth` parameter is optional and allows you to use registries that are not public, and must be accessed through an authentication.
```

Author: Phrase modifiée

The parameter is the way for you to tell composer how to authenticate to the registry server.

Each auth can have the following configuration fields:
- `url`: the url of your repository
- `token`: when you use a connection via token, you must use this field

> > Notice : Currently, the only way to identify to a repository is to use tokens. Support for other authentication methods is in our backlog.

Reviewer: le mettre après le block de code ?

Author: C'est fait
